### PR TITLE
Bug 1201615 - Give swagger the correct base URL

### PR DESCRIPTION
**Pull request by smarterclayton**

### CODE REVIEW

**Comment by smarterclayton** [comment]
@deads2k review

File: pkg/cmd/server/api/types.go [hunk 1]
```diff
@@ -47,10 +47,14 @@ type MasterConfig struct {
 	// CORSAllowedOrigins
 	CORSAllowedOrigins []string
 
+	// MasterPublicURL is how clients can access the OpenShift API server
+	MasterPublicURL string
```
Comment (deads2k) [inline pkg/cmd/server/api/types.go]: Add validation to ensure the URL is present (required for swagger), well-formed, and matches any other MasterPublicURLs that may be present.

Comment (deads2k) [inline pkg/cmd/server/api/types.go]: Oh, and for the next question: each stanza has its own copy to keep it contained if we decide to allow starting these components separately.

Comment (deads2k) [inline pkg/cmd/server/api/types.go]: @liggitt `PublicURL` to match the `AssetConfig`?

Comment (smarterclayton) [inline pkg/cmd/server/api/types.go]: We don't validate well formedness of the others.

Also, why do they have to match?

Comment (smarterclayton) [inline pkg/cmd/server/api/types.go]: MasterConfig has no effective stanza.  What would you suggest.

Comment (liggitt) [inline pkg/cmd/server/api/types.go]: > `PublicURL` to match the `AssetConfig`?

I can't decide whether I like everything using the same names to refer to the same things, or removing prefixes based on context. I think I lean toward the same names... but I don't feel that strongly

Comment (smarterclayton) [inline pkg/cmd/server/api/types.go]: Same name.  Less mental struggle.

----- Original Message -----

**Comment by deads2k** [comment]
lgtm [merge]

File: pkg/cmd/server/api/types.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/server/api/types.go]: We're not validating those URLs today.  Why?

----- Original Message -----
> We don't validate well formedness of the others.
> 
> Also, why do they have to match?
>

Comment (deads2k) [inline pkg/cmd/server/api/types.go]: > We're not validating those URLs today.  Why?

Oversight?

**Comment by smarterclayton** [comment]
Added unit tests.

**Comment by deads2k** [comment]
[merge]

**Pull request merged**